### PR TITLE
Implemented TripDetailsViewController

### PR DESCRIPTION
**Pull request by rs929**
## Overview

Implemented View for TripDetailsViewController

## Changes Made

- Added assets for all icons on the page
- Implemented all views for TripDetailsViewController
- 
## Test Coverage

- Tested Views on screens of multiple sizes
- Positioning of Views is relative

## Next Steps (delete if not applicable)

- Implement Request rides networking

## Screenshots (delete if not applicable)

Screenshot of TripDetailsViewController:
Simulator Screen Shot - iPhone 14 Pro - 2023-03-11 at 00 19 40

### CODE REVIEW

**Review by tiffany-pann — APPROVED** [review]
LGTM! Just one minor thing, I noticed that the StackView makes use of many subviews that you setup directly in the `setUpStackView()` function.  Is it possible for you to refactor some of that code out into smaller functions like we do with other views that we build? Just for easier debugging/readability in the future - but I'm also not too familiar w. stack views so maybe that's not possible. Good work Richie!

**Review by tiffany-pann — APPROVED** [review]

File: Scoop/Controller/TripDetailsViewController.swift [hunk 1]
```diff
@@ -26,36 +30,306 @@ class TripDetailsViewController: UIViewController {
...
-        creatorLabel.text = creator.firstName
-        stackView.addArrangedSubview(creatorLabel)
+        creatorProfile.image = UIImage(named: "emptyimage") //TODO: Replaces once Backend finishes
+        creatorProfile.contentMode = .scaleAspectFill
+        creatorProfile.layer.borderWidth = 1
+        creatorProfile.layer.masksToBounds = false
+        creatorProfile.layer.borderColor = UIColor.black.cgColor
+        creatorProfile.layer.cornerRadius = 20.5
+        creatorProfile.clipsToBounds = true
+        driverInfoContainerView.addSubview(creatorProfile)
```
Comment (ReadeP) [inline Scoop/Controller/TripDetailsViewController.swift:98]: newline

File: Scoop/Controller/TripDetailsViewController.swift [hunk 2]
```diff
@@ -26,36 +30,306 @@ class TripDetailsViewController: UIViewController {
...
+        creatorProfile.snp.makeConstraints { make in
+            make.size.equalTo(41)
+            make.leading.equalToSuperview()
+            make.centerY.equalToSuperview()
+        }
+        
+        creatorLabel.text = "\(creator.firstName) \(creator.lastName)"
+        driverInfoContainerView.addSubview(creatorLabel)
```
Comment (ReadeP) [inline Scoop/Controller/TripDetailsViewController.swift:107]: newline

File: Scoop/Controller/TripDetailsViewController.swift [hunk 3]
```diff
@@ -26,36 +30,306 @@ class TripDetailsViewController: UIViewController {
...
+        creatorLabel.snp.makeConstraints { make in
+            make.leading.equalTo(creatorProfile.snp.trailing).inset(-iconTextSpacing)
+            make.top.equalTo(creatorProfile)
+        }
+        
+        driverInfoContainerView.addSubview(mailIcon)
```
Comment (ReadeP) [inline Scoop/Controller/TripDetailsViewController.swift:114]: newline

File: Scoop/Controller/TripDetailsViewController.swift [hunk 4]
```diff
@@ -26,36 +30,306 @@ class TripDetailsViewController: UIViewController {
...
+        mailIcon.snp.makeConstraints { make in
+            make.size.equalTo(15)
+            make.leading.equalTo(creatorLabel)
+            make.bottom.equalTo(creatorProfile)
+        }
+        
+        creatorEmail.text = "\(creator.netid)@cornell.edu"
+        driverInfoContainerView.addSubview(creatorEmail)
```
Comment (ReadeP) [inline Scoop/Controller/TripDetailsViewController.swift:123]: newline

File: Scoop/Controller/TripDetailsViewController.swift [hunk 5]
```diff
@@ -26,36 +30,306 @@ class TripDetailsViewController: UIViewController {
...
+        creatorEmail.snp.makeConstraints { make in
+            make.top.equalTo(creatorLabel.snp.bottom)
+            make.leading.equalTo(mailIcon.snp.trailing).inset(-iconTextSpacing)
+        }
+        
+        stackView.addArrangedSubview(driverInfoContainerView)
```
Comment (ReadeP) [inline Scoop/Controller/TripDetailsViewController.swift:130]: newline

File: Scoop/Controller/TripDetailsViewController.swift [hunk 6]
```diff
@@ -26,36 +30,306 @@ class TripDetailsViewController: UIViewController {
...
+            make.height.equalTo(41)
+        }
+        
+    }
+    
+    private func setUpTransportationInfo() {
+        guard let ride = currentRide else { return }
+        
+        transportationMethod.text = "TRANSPORTATION METHOD"
+        transportationContainerView.addSubview(transportationMethod)
```
Comment (ReadeP) [inline Scoop/Controller/TripDetailsViewController.swift:142]: separate view.addSubview and constraints with a newline for better readability

File: Scoop/Controller/TripDetailsViewController.swift [hunk 7]
```diff
@@ -26,36 +30,306 @@ class TripDetailsViewController: UIViewController {
...
+            make.top.equalTo(stackView.snp.bottom)
+            make.centerX.equalToSuperview()
+            make.height.equalTo(51)
+            make.width.equalTo(296)
+        }
+    }
+    
+    @objc private func requestRide() {
+        //TODO: Networking Goes here
+        self.dismiss(animated: true)
```
Comment (ReadeP) [inline Scoop/Controller/TripDetailsViewController.swift]: remove `self`

**Review by ReadeP — APPROVED** [review]
Looks good!!!

**Pull request merged**